x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/inference/bulk/BulkInferenceExecutionConfig.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.inference.bulk;
+
+public record BulkInferenceExecutionConfig(int workers, int maxOutstandingRequests) {
```

Comment:
We don't plan to make these configurable in the near future - do we ever use anything else than the `DEFAULT`?
I am not a fan of adding these types of record classes that are only used to store defaults that are not configurable.
To me this just adds a cognitive load for anyone looking into rerank/completion.
I'd rather have the `DEFAULT_WORKERS` and `DEFAULT_MAX_OUTSTANDING_REQUESTS` in the base operator that uses them, instead of carrying around these record objects.

Maybe we anticipate these will be configurable, but until then this serves no purpose. I am a huge believer that we shouldn't add abstractions/constructions like these until they are actually used.

Reply:
it's okay - we don't need to block merging this PR because of this one